src/logging.h

```diff
@@ -167,4 +168,50 @@ static inline void LogPrint(const BCLog::LogFlags& category, const Args&... args
     }
 }
 
+class ILogFilter
+{
+public:
+    virtual ~ILogFilter(){};
+    virtual bool filter() = 0;
+};
+
+// Implementation that logs at most every millis_ milliseconds
+class TimeThrottledFilter : public ILogFilter
+{
+private:
+    uint64_t millis_{0};
+    uint64_t last_time_millis_{0};
+
+public:
+    TimeThrottledFilter() = delete;
+    explicit TimeThrottledFilter(uint64_t millis) : millis_(millis)
+    {
+    }
+
+    virtual bool filter() override
+    {
+        int64_t current_time = GetTimeMillis();
+        // First call or at least millis_ ms to last call
+        if ((last_time_millis_ == 0) || ((current_time - last_time_millis_) > millis_)) {
+            last_time_millis_ = current_time;
+            return true;
+        }
+        return false;
```

Comment:
Reverse logic here. When i call function called filter is expect to understand is something to be filtered or not. So `true` means should be filtered

Reply:
Done